fsspec/spec.py

```diff
@@ -886,13 +886,21 @@ def get(self, rpath, lpath, recursive=False, callback=_DEFAULT_CALLBACK, **kwarg
             trailing_sep_maybe_asterisk,
         )
 
+        rpaths = self.expand_path(rpath, recursive=recursive)
+        if (
+            len(rpaths) == 1
+            and not recursive
+            and (trailing_sep(rpaths[0]) or self.isdir(rpaths[0]))
+        ):
+            # Non-recursive copy of directory does nothing.
+            return
+        source_is_str = isinstance(rpath, str)
+
```

Comment:
I have reordered the code here (and the other 2 functions below) to separate out the handling of `source` and `target` paths, with a line of whitespace between, for increased clarity.